src/components/swap/PendingModalContent/Step.tsx

```diff
@@ -0,0 +1,193 @@
+  width: 24px;
+  height: 24px;
+  filter: ${({ isActive }) => `grayscale(${isActive ? 0 : 1})`};
+  opacity: ${({ isActive }) => (isActive ? '1' : '0.5')};
+`
+function RippleAnimation({ rippleColor }: { rippleColor?: string }) {
+  if (!rippleColor) {
+    return null
+  }
+  return (
+    <div data-testid="icon-ripple-animation">
+      <Ring width="1px" color={rippleColor} animation={innerRingAnimation} />
+      <Ring width="0.5px" color={rippleColor} animation={outerRingAnimation} />
+    </div>
+  )
+}
+
+function Icon({ stepStatus, icon, rippleColor }: { stepStatus: StepStatus; icon: ReactElement; rippleColor?: string }) {
+  if (stepStatus === StepStatus.IN_PROGRESS) {
+    return <LoaderV3 size="24px" fill={rippleColor} data-testid="loader-icon" />
+  }
+  return (
+    <div>
+      {stepStatus === StepStatus.ACTIVE && <RippleAnimation rippleColor={rippleColor} />}
+      <IconWrapper isActive={stepStatus === StepStatus.ACTIVE} data-testid="step-icon">
+        {icon}
+      </IconWrapper>
+    </div>
+  )
+}
+
+function Title({
+  stepStatus,
+  stepDetails,
+  isTimeRemaining,
+}: {
+  stepStatus: StepStatus
+  stepDetails: StepDetails
+  isTimeRemaining: boolean
+}) {
+  switch (stepStatus) {
+    case StepStatus.PREVIEW:
+      return <ThemedText.BodySecondary>{stepDetails.previewTitle}</ThemedText.BodySecondary>
+    case StepStatus.ACTIVE:
+      return isTimeRemaining ? (
+        <ThemedText.BodyPrimary>{stepDetails.actionRequiredTitle}</ThemedText.BodyPrimary>
+      ) : (
+        <ThemedText.BodyPrimary>{stepDetails.delayedStartTitle}</ThemedText.BodyPrimary>
+      )
+    case StepStatus.IN_PROGRESS:
+      return isTimeRemaining ? (
+        <ThemedText.BodyPrimary>{stepDetails.inProgressTitle}</ThemedText.BodyPrimary>
+      ) : (
+        <ThemedText.BodyPrimary>{stepDetails.delayedEndTitle}</ThemedText.BodyPrimary>
+      )
+    case StepStatus.COMPLETE:
+      return <ThemedText.BodySecondary>{stepDetails.previewTitle}</ThemedText.BodySecondary>
+    default:
+      return null
+  }
+}
+
+function Timer({ secondsRemaining }: { secondsRemaining: number }) {
+  const minutes = Math.floor(secondsRemaining / 60)
+  const seconds = secondsRemaining % 60
+  const minutesText = minutes < 10 ? `0${minutes}` : minutes
+  const secondsText = seconds < 10 ? `0${seconds}` : seconds
+  const timerText = `${minutesText}:${secondsText}`
+  return (
+    <ThemedText.BodySecondary paddingRight="8px" data-testid="step-timer">
+      {timerText}
+    </ThemedText.BodySecondary>
+  )
+}
+
+export function Step({ stepStatus, stepDetails }: { stepStatus: StepStatus; stepDetails: StepDetails }) {
+  // Timer is shown in three cases:
+  // (1) User has a specified amount of time to perform a required action. Timer starts running as soon as the step becomes active.
+  // (2) Step has an estimated amount of time in which it should be completed. Timer is set but not running when step becomes active.
```

Comment:
i think we should still hide the timer until the step is in progress, it seems strange to me to have the timer in view and paused while waiting for user input

Reply:
I was thinking of it like a shot clock in sports; always remains on but doesn't start until activity begins. I agree with you though that it's simpler to just hide the timer until it's running. I've made the changes and also simplified the time-keeping logic.